tests/integration/snowflake/test_snowflake_adapter.py

```diff
@@ -0,0 +1,349 @@
+import urllib.parse
+from random import randrange
+
+import pytest
+import yaml
+
+import snowshu.core.models.materializations as mz
+from snowshu.adapters.source_adapters import BaseSourceAdapter
+from snowshu.adapters.source_adapters.snowflake_adapter import SnowflakeAdapter
+from snowshu.core.models.credentials import Credentials
+from snowshu.core.models.relation import Relation
+from snowshu.exceptions import TooManyRecords
+from snowshu.samplings.sample_methods import BernoulliSampleMethod
+from tests.assets.integration_test_setup import CREDENTIALS, get_connection_profile
+from tests.common import query_equalize, rand_string
+
+
+@pytest.fixture(scope='session')
+def sf_adapter():
+    with open(CREDENTIALS) as cred_file:
+        credentials = yaml.safe_load(cred_file)
+
+    profile_dict = get_connection_profile(credentials)
+    profile_dict.update({"role": "snowshu_development_role"})
```

Comment:
It's probably worth enforcing the role value to be present in the credentials file.

Reply:
In that case it may be worth mentioning somewhere that: 

> You should use snowshu_development_user/role when running integration tests